runtime/util/vmargs.c

```diff
@@ -76,32 +76,37 @@
 
 #define LARGE_STRING_BUF_SIZE 256
 
+#define JAVA_CLASS_PATH_EQUALS "-Djava.class.path="
+#define SUN_JAVA_COMMAND_EQUALS "-Dsun.java.command="
+#define SUN_JAVA_LAUNCHER_PID_EQUALS "-Dsun.java.launcher.pid="
+#define SUN_JAVA_LAUNCHER_EQUALS "-Dsun.java.launcher="
+
 struct JavaLauncherOption {
 	char *optionName;
 	int optionNameLength;
 };
 
-/* options in this table will be not treated as a user Argument 
-Note, these options need to be in lexical order (ASCII values) for the binary search function to properly work */
+/* Options in this table will be not treated as user arguments.
+Note, these options need to be in lexical order (ASCII values) for the binary search function to work properly */
 static const struct JavaLauncherOption JavaLauncherOptionTable[] = {
-	{"-Djava.class.path=", strlen("-Djava.class.path=")},
-	{"-Dsun.java.command=", strlen("-Dsun.java.command=")},
-	{"-Dsun.java.launcher.pid=", strlen("-Dsun.java.launcher.pid=")},
-	{"-Dsun.java.launcher=", strlen("-Dsun.java.launcher=")},/* Note, 'launcher=' is placed after .launcher.pid` since '=' has a higher ASCII value than '.' */
-	{"-Xjcl:", strlen("-Xjcl:")},
+	{JAVA_CLASS_PATH_EQUALS, strlen(JAVA_CLASS_PATH_EQUALS)},
+	{SUN_JAVA_COMMAND_EQUALS, strlen(SUN_JAVA_COMMAND_EQUALS)},
+	{SUN_JAVA_LAUNCHER_PID_EQUALS, strlen(SUN_JAVA_LAUNCHER_PID_EQUALS)},
+	{SUN_JAVA_LAUNCHER_EQUALS, strlen(SUN_JAVA_LAUNCHER_EQUALS)},/* Note, 'launcher=' is placed after .launcher.pid` since '=' has a higher ASCII value than '.' */
+	{VMOPT_XJCL_COLON, strlen(VMOPT_XJCL_COLON)}, /* macro from jvminit.h */
 };
 
 #define ignoredOptionTableEnd (sizeof(JavaLauncherOptionTable) / sizeof(struct JavaLauncherOption) - 1)
 
 static char * getStartOfOptionValue(J9VMInitArgs* j9vm_args, IDATA element, const char *optionName);
 
 /* Binary Search function that searches if optionName is in JavaLauncherOptionTable */
-UDATA
+int
 isGeneratedArgument(const char *optionName) {
-	IDATA start = 0;
-	IDATA end = ignoredOptionTableEnd;
-	IDATA mid = 0;
-	IDATA compareResults = 0;
+	int start = 0;
```

Comment:
Check with GAC or Keith as to the preferred type in this case.

Reply:
I was recommended to use UDATA

Reply:
I was recommended to use UDATA